calyx/src/analysis/static_par_timing.rs

```diff
@@ -0,0 +1,192 @@
+use crate::{
+    analysis::ControlId,
+    ir::{self},
+};
+use std::{
+    collections::{HashMap, HashSet},
+    fmt::Debug,
+};
+
+/// maps the node ids of enables to a set of tuples (i,j), which is clock cycles (relative
+/// to the start of the par) that enable is live
+type ParTimingMap = HashMap<u64, HashSet<(u64, u64)>>;
```

Comment:
There might be a smarter way to represent the inner HashSet; for example, if we sort the ranges, we can perform a binary search the list to figure out when something is active.

In general, maybe can think about a nicer interface for ParTimingMap to the outside world?